src/source/Signaling/ChannelInfo.c

```diff
@@ -64,7 +64,7 @@ STATUS createValidateChannelInfo(PChannelInfo pOrigChannelInfo, PChannelInfo* pp
         pRegionPtr = DEFAULT_AWS_REGION;
     }
 
-    if (pOrigChannelInfo->pControlPlaneUrl != NULL) {
+    if (pOrigChannelInfo->pControlPlaneUrl != NULL && *pOrigChannelInfo->pControlPlaneUrl != '\0') {
```

Comment:
minor: we can use

https://github.com/awslabs/amazon-kinesis-video-streams-pic/blob/c98c2a256a7bb3dfc4db41ae26d45e28ac7eec56/src/common/include/com/amazonaws/kinesis/video/common/CommonDefs.h#L906